determined-ai/determined
# feat: Add warning for submissions for requests that cannot currently be fulfilled [DET-6410]

## Description

feat: Add warning for submissions for requests that cannot currently be fulfilled [DET-6410]

Presently, if a user requests a job with a larger number of slots than are currently available the job may be stuck in queue indefinitely. This PR shows a **warning** message in the CLI and WebUI to a user letting them know that too many slots may have been requested. 

**important note:** Since we cannot accurately tell if more slots may become available in the future we are **not** blocking job submission, only showing a warning to the user.

![Screen Shot 2022-11-03 at 10 04 00 AM](https://user-images.githubusercontent.com/103522725/199757599-bd83d3db-9161-4dc6-9148-88cfdd2bdb8d.png)

<img width="1501" alt="Screen Shot 2022-11-03 at 10 03 26 AM" src="https://user-images.githubusercontent.com/103522725/199757470-0a091208-3556-4ac5-9d6a-6641dcc1dcb6.png">

![Screen Shot 2022-11-03 at 10 33 32 AM](https://user-images.githubusercontent.com/103522725/199764898-e3070530-f446-4432-a4d8-ed79ede66f65.png)





## Test Plan

1. Attempt to create a notebook in the WebUI requesting more slots than available. Ensure that you see a warning on the about the number of slots. 
2. Attempt to create a notebook in the CLI requesting more slots than available. Ensure that you see a warning about the number of slots. 
3. Attempt to fork an experiment and in the WebUI and in the configuration specify more slots than available. Ensure that you see a warning about the number of slots. 
4. Create an experiment in the CLI that requires more slots than available. Ensure that you see a warning about the number of slots. 
5. Repeat steps 1-4 instead specify a number a slots equal to or less than the amount available and ensure that you do not see any warning messages. 



## Commentary (optional)




## Checklist
~- [ ] User-facing API changes need the "User-facing API Change" label.~
~- [ ] Release notes should be added as a separate file under `docs/release-notes/`.~
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
~- [ ] Licenses should be included for new code which was copied and/or modified from any external code.~





[DET-6410]: https://determinedai.atlassian.net/browse/DET-6410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
